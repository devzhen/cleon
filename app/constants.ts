export const DEFAULT_COORDS = {
  top: 100,
  left: 220,
} as const;

export const BOARD_TYPE = {
  left: 'left',
  right: 'right',
} as const;