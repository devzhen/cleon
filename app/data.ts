import { v4 as uuid } from "uuid";

export const stickers = [
  {
    id: uuid(),
    top: 110,
    left: 230,
    createdAt: new Date().toISOString(),
    text: ''
  }
];
