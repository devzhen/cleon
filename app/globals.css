:root {
  --color-active-board: #eb6529;
  --color-not-active-board: #bfc3ca;
  --color-main-background: #f0cc46;
  --color-board-background: #e9ecf1;
  --color-header-right-side: #7b7b7b;
  --color-modal-background: #fff8c8;
  --color-text-area-background: #fffde6;
  --color-text-area-border: #dfe2e7;
  --color-text-area-color: #2e2e2e;
  --color-date-color: #979797;
}

html,
body {
  max-width: 100vw;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.main {
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;
}

.boardActive {
  border-color: var(--color-active-board) !important;
}

.boardNotActive {
  border-color: var(--color-not-active-board) !important;
}
