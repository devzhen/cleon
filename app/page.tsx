"use client";

import styled from "styled-components";

import Header from "./components/Header";
import LeftSide from "./components/LeftSide";

export default function Home() {
  return (
    <Main>
      <header>
        <Header />
      </header>
      <LeftSide />
    </Main>
  );
}

const Main = styled.main`
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;

  header {
    grid-column: 2 span;
    height: 66px;
    width: 100%;
  }
`;
