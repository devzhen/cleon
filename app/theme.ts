const theme = {
  colors: {
    activeBoard: '#eb6529',
    notActiveBoard: '#bfc3ca',
    headerTopBackground: '#363636',
    mainBackground: '#f0cc46',
    boardBackground: '#e9ecf1',
    headerLogoTitle: '#232323',
    headerRightSide: '#7b7b7b',
    white: '#fff',
    modalBackground: '#fff8c8',
    textAreaBackground: '#fffde6',
    textAreaBorder: '#dfe2e7',
    textAreaColor: '#2e2e2e',
    dateColor: '#979797',
  },
} as const;

export default theme;
