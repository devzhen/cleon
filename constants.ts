export const DEFAULT_STICKER_POS = {
  top: 100,
  left: 220,
} as const;

export const BOARD_TYPE = {
  left: 'left',
  right: 'right',
  all: 'all',
} as const;

export const STICKERS_INITIAL_VALUE = { [BOARD_TYPE.left]: {}, [BOARD_TYPE.right]: {}};
