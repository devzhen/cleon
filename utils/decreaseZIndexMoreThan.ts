import { BOARD_TYPE } from '@/constants';

const decreaseZIndexMoreThan = (
  board: (typeof BOARD_TYPE)[keyof typeof BOARD_TYPE],
  zIndex: number,
) => {
  const stickers = document.querySelectorAll(`.sticker[data-board="${board}"`);

  for (let i = 0; i < stickers.length; i++) {
    const sticker = stickers[i] as HTMLDivElement;

    const stickerZIndex = parseInt(window.getComputedStyle(sticker, null).zIndex);

    if (stickerZIndex >= zIndex) {
      sticker.style.zIndex = `${Math.max(0, stickerZIndex - 1)}`;
    }
  }
};

export default decreaseZIndexMoreThan;
