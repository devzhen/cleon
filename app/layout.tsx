import type { Metadata } from "next";
import { Roboto } from "next/font/google";

import "./globals.css";

import StyledComponentsRegistry from "../lib/registry";

const roboto = Roboto({
  weight: ["400", "700"],
  style: ["normal"],
  subsets: ["latin"],
  display: "swap",
});

export const metadata: Metadata = {
  title: "Cleon",
  description: "Cleon",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={roboto.className}>
        <StyledComponentsRegistry>{children}</StyledComponentsRegistry>
      </body>
    </html>
  );
}
