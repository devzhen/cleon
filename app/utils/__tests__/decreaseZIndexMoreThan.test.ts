import { BOARD_TYPE } from '@/app/constants';

import decreaseZIndexMoreThan from '../decreaseZIndexMoreThan';

describe('decreaseZIndexMoreThan', () => {
  it('without stickers', () => {
    decreaseZIndexMoreThan(BOARD_TYPE.left, 1);

    expect(document.body).toMatchSnapshot();
  });

  it('with stickers', () => {
    const sticker1 = document.createElement('div');
    sticker1.setAttribute('data-board', BOARD_TYPE.left);
    sticker1.style.zIndex = '2';
    sticker1.classList.add('sticker');
    document.body.appendChild(sticker1);

    const sticker2 = document.createElement('div');
    sticker2.setAttribute('data-board', BOARD_TYPE.left);
    sticker2.style.zIndex = '3';
    sticker2.classList.add('sticker');
    document.body.appendChild(sticker2);

    decreaseZIndexMoreThan(BOARD_TYPE.left, 2);

    expect(document.body).toMatchSnapshot();
  });
});
