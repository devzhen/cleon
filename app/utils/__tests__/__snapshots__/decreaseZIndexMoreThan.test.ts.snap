// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decreaseZIndexMoreThan with stickers 1`] = `
<body>
  <div
    class="sticker"
    data-board="left"
    style="z-index: 1;"
  />
  <div
    class="sticker"
    data-board="left"
    style="z-index: 2;"
  />
</body>
`;

exports[`decreaseZIndexMoreThan without stickers 1`] = `<body />`;
